pallets/providers/src/utils.rs

```diff
@@ -1,34 +1,237 @@
-        Ok(())
+        // todo!("If this comment is present, it means this function is still incomplete even though it compiles.")
+
+        // Check that, by registering this Backup Storage Provider, we are not exceeding the maximum number of Backup Storage Providers
+        let new_amount_of_bsps = BspCount::<T>::get()
+            .checked_add(&T::SpCount::one())
+            .ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+        ensure!(
+            new_amount_of_bsps <= T::MaxBsps::get(),
+            Error::<T>::MaxBspsReached
+        );
+
+        // Check that the account is not already registered either as a Main Storage Provider or a Backup Storage Provider
+        ensure!(
+            AccountIdToMainStorageProviderId::<T>::get(who).is_none()
+                && AccountIdToBackupStorageProviderId::<T>::get(who).is_none(),
+            Error::<T>::AlreadyRegistered
+        );
+
+        // Check that the multiaddresses vector is not empty (SPs have to register with at least one)
+        ensure!(
+            !bsp_info.multiaddresses.is_empty(),
+            Error::<T>::NoMultiAddress
+        );
+
+        // Check that the multiaddresses are valid
+        for multiaddress in bsp_info.multiaddresses.iter() {
+            let multiaddress_vec = multiaddress.to_vec();
+            let valid_multiaddress = Multiaddr::try_from(multiaddress_vec);
+            match valid_multiaddress {
+                Ok(_) => (),
+                Err(_) => return Err(Error::<T>::InvalidMultiAddress.into()),
+            }
+        }
+
+        // Check that the data to be stored is bigger than the minimum required by the runtime
+        ensure!(
+            bsp_info.capacity >= T::SpMinCapacity::get(),
+            Error::<T>::StorageTooLow
+        );
+
+        // Calculate how much deposit will the signer have to pay to register with this amount of data
+        let capacity_over_minimum = bsp_info
+            .capacity
+            .checked_sub(&T::SpMinCapacity::get())
+            .ok_or(Error::<T>::StorageTooLow)?;
+        let deposit_for_capacity_over_minimum = T::DepositPerData::get()
+            .checked_mul(&capacity_over_minimum.into())
+            .ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+        let deposit = T::SpMinDeposit::get()
+            .checked_add(&deposit_for_capacity_over_minimum)
+            .ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+
+        // Check if the user has enough balance to pay the deposit
+        let user_balance =
+            T::NativeBalance::reducible_balance(who, Preservation::Preserve, Fortitude::Polite);
+        ensure!(user_balance >= deposit, Error::<T>::NotEnoughBalance);
+
+        // Check if we can hold the deposit from the user
+        ensure!(
+            T::NativeBalance::can_hold(&HoldReason::StorageProviderDeposit.into(), who, deposit),
+            // TODO: Change this error to something more descriptive
```

Comment:
I actually wanted to do some more research on the reasons `can_hold` can fail (other than not having enough balance), but I can set up a more generic `CantHoldRequiredBalance` error (which makes sense, since I'm not going to map `can_hold` errors one by one)